packages/usa-date-picker/src/index.js

```diff
@@ -464,6 +464,37 @@ const isDatesYearOutsideMinOrMax = (date, minDate, maxDate) =>
   lastDayOfMonth(setMonth(date, 11)) < minDate ||
   (maxDate && startOfMonth(setMonth(date, 0)) > maxDate);
 
+/**
+ * @typedef {Object} DateRangeContext
+ * @property {Date} rangeStartDate
+ * @property {Date} rangeEndDate
+ * @property {Date} withinRangeStartDate
+ * @property {Date} withinRangeEndDate
+ */
```

Comment:
**praise:** Thanks for adding this, I see the definitions on hover in VS Code.

![image](https://github.com/user-attachments/assets/832510a2-6580-469c-b5a2-629a8ce4b0c8)